File: Java/src/com/leetcode/Medium/MajorityElementII.java
package com.leetcode.Medium;

import java.util.*;

// 229. Majority Element II
public class MajorityElementII {
    public List<Integer> majorityElement(int[] nums) {
        int n = nums.length / 3;
        List<Integer> res = new ArrayList<>();
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums)
            map.put(num, map.getOrDefault(num, 0) + 1);

        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() > n)
                res.add(entry.getKey());
        }
        return res;
    }
}

File: JavaScript/Medium/MajorityElementII.js
/**
 * 229. Majority Element II
 * @param {number[]} nums
 * @return {number[]}
 */
const majorityElement = function (nums) {
  let n = Math.floor(nums.length / 3);
  let res = [];
  let map = new Map();

  for (let num of nums) {
    map.set(num, (map.get(num) || 0) + 1);
  }

  for (let [key, value] of map) {
    if (value > n)
      res.push(key);
  }

  return res;
};